### ISSUE TITLE
Update `@yamada-ui/menu` accessibility

### ISSUE THREAD

**Issue by nakayan5** [yamada-ui/yamada-ui#1923]
### Description

This issue aims to address accessibility deficiencies in our button component as per W3C guidelines.
The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

## menubutton

- [ ] Optionally, the element with role button has a value specified for [aria-controls](https://w3c.github.io/aria/#aria-controls) that refers to the element with role menu.

## menu

Enter:

  - [ ] When focus is on a menuitem that has a submenu, opens the submenu and places focus on its first item.
 Otherwise, activates the item and closes the menu.

Down Arrow:

 - [x] When focus is on a menuitem in a menubar, and that menuitem has a submenu, opens the submenu and places focus on the first item in the submenu.

- [x] If activating a [menuitem](https://w3c.github.io/aria/#menuitem) opens a submenu, the menuitem is known as a parent menuitem. A submenu's menu element is:
  - [x] Contained inside the same menu element as its parent menuitem.
  - [x] Is the sibling element immediately following its parent menuitem.

- [ ] When a [menuitemcheckbox](https://w3c.github.io/aria/#menuitemcheckbox) or [menuitemradio](https://w3c.github.io/aria/#menuitemradio) is checked, [aria-checked](https://w3c.github.io/aria/#aria-checked) is set to true.

- [ ] All [separators](https://w3c.github.io/aria/#separator) should have [aria-orientation](https://w3c.github.io/aria/#aria-orientation) consistent with the separator's orientation.

- [x]  If a menubar has a visible label, the element with role menubar has [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set to a value that refers to the labelling element. Otherwise, the menubar element has a label provided by [aria-label](https://w3c.github.io/aria/#aria-label).

- [x]  If a menubar is vertically oriented, it has [aria-orientation](https://w3c.github.io/aria/#aria-orientation) set to vertical. The default value of aria-orientation for a menubar is horizontal.
- [x]  An element with role menu either has:
  - [x]  [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set to a value that refers to the menuitem or button that controls its display.
- [x]  A label provided by [aria-label](https://w3c.github.io/aria/#aria-label).
- [x] If a menu is horizontally oriented, it has [aria-orientation](https://w3c.github.io/aria/#aria-orientation) set to horizontal. The default value of aria-orientation for a menu is vertical.

### Problem Statement/Justification

improve accessibility

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

**Comment by nakayan5** [comment]
related https://github.com/yamada-ui/yamada-ui/issues/1501
related https://github.com/yamada-ui/yamada-ui/issues/1502

**Comment by taiseiiiii** [comment]
> When focus is on a menuitem in a menubar, and that menuitem has a submenu, opens the submenu and places focus on the first item in the submenu.

This issue is not related to menubar. So this checkpoint exclude.

**Comment by taiseiiiii** [comment]
> メニューバーに表示ラベルがある場合、ロール menubar を持つ要素には、ラベル付け要素を参照する値に設定された[aria-labelledbyがあります。それ以外の場合、メニューバー要素には](https://w3c.github.io/aria/#aria-labelledby)[aria-label](https://w3c.github.io/aria/#aria-label)によって提供されるラベルがあります。

This issue is not related to menubar. So this checkpoint exclude too.

**Comment by taiseiiiii** [comment]
> A label provided by [aria-label](https://w3c.github.io/aria/#aria-label).

Above checkpoint is unnecessary.  Because `aria-labelledby` attribute is already set in the component which has `role` of `menu`

According to the following text, only one of these is required.
https://www.w3.org/WAI/ARIA/apg/patterns/menubar/
> An element with role menu either has:
> [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set to a value that refers to the menuitem or button that controls its display.
> A label provided by [aria-label](https://w3c.github.io/aria/#aria-label).

**Comment by taiseiiiii** [comment]
> If a menu is horizontally oriented, it has [aria-orientation](https://w3c.github.io/aria/#aria-orientation) set to horizontal. The default value of aria-orientation for a menu is vertical.

Yamada UI only provides horizontal menu, so above checkpoint is unnecessary.

**Comment by taiseiiiii** [comment]
> If activating a [menuitem](https://w3c.github.io/aria/#menuitem) opens a submenu, the menuitem is known as a parent menuitem. A submenu's menu element is:
> 
>  Contained inside the same menu element as its parent menuitem.
>  Is the sibling element immediately following its parent menuitem.

Above checkpoint is unnecessary.
Because `Menu` component is implemented with proper parent-child relationships.
https://yamada-ui.com/components/overlay/menu
